src/components/videos/videoCard.js

```diff
@@ -1,41 +1,47 @@
 import { Link } from "react-router-dom";
 import { AiFillLike } from "react-icons/ai";
 import { MdWatchLater, MdPlaylistAdd } from "react-icons/md";
-import { useLike, useWatchLater } from "../../contexts/allContext";
-
-export const VideoCard = (video) => {
+import { useLike, useWatchLater, useHistory, usePlaylist, useVideo } from "../../contexts/allContext";
+import { CreatePlaylist } from "../playlist/createPlaylist";
+import { useState } from "react"
+export const VideoCard = (v) => {
     const { LikeVideoHandler, videoState: { liked }, RemoveLikeVideoHandler } = useLike()
     const { videoState: { watchLater }, AddWatchLaterHandler, RemoveWatchLaterHandler } = useWatchLater()
-    const { _id, title, creator, views } = video;
+    const { HistoryVideoHandler } = useHistory()
+    const { _id, title, creator, views } = v;
+    const { playListM, showPlayListM } = usePlaylist()
+    const { video } = useVideo()
```

Comment:
I think you can remove line 13 as you are using v from props for video details.

Reply:
Yes thank you, I will do that